beacon-chain/rpc/validator/status.go

```diff
@@ -194,6 +184,7 @@ func (vs *Server) assignmentStatus(validatorIdx uint64, beaconState *stateTrie.B
 func (vs *Server) depositBlockSlot(ctx context.Context, eth1BlockNumBigInt *big.Int, beaconState *stateTrie.BeaconState) (uint64, error) {
 	var depositBlockSlot uint64
 	blockTimeStamp, err := vs.BlockFetcher.BlockTimeByHeight(ctx, eth1BlockNumBigInt)
+	// This is the main expensive RPC call, how can this be improved?
```

Comment:
You may want to consider a cache if this data is fetched repeatedly for the same validator.  There is an implementation at https://github.com/prysmaticlabs/prysm/blob/master/beacon-chain/rpc/beacon/validators_stream.go#L493-L509

Reply:
Very good idea, thank you!

Reply:
Will address this in another PR, #5340 to track